lib/koans/18_genservers.ex

```diff
@@ -0,0 +1,68 @@
+defmodule GenServers do
+  use Koans
+
+  @intro "GenServers"
+
+  defmodule ExampleGenServer do
+    use GenServer
+
+    # Use init to set up initial state []
+    def init(_args) do
+      {:ok, []}
+    end
+
+    # Use calls to perform synchronous actions
+    def handle_call(:pop, _from, [head|tail]) do
+      {:reply, head, tail}
+    end
+
+    def handle_call(:peek, _from, stack) do
+      {:reply, stack, stack}
+    end
+
+    # Use casts to perform asynchronous actions
+    def handle_cast({:push, value}, stack) do
+      {:noreply, [value|stack]}
+    end
+  end
+
+
+  koan "You can think of GenServers like more flexible agents" do
+    {:ok, pid} = GenServer.start_link(ExampleGenServer, :ok, [])
+
+    assert is_pid(pid) == ___
+  end
+
+  koan "GenServers can have a name too" do
+    {:ok, pid} = GenServer.start_link(ExampleGenServer, :ok, name: StackServer)
+
+    assert is_pid(GenServer.whereis(StackServer)) == ___
+  end
+
+  koan "Use call to send a synchronous request to the server" do
+    GenServer.start_link(ExampleGenServer, :ok, name: StackServer)
+
+    stack = GenServer.call(StackServer, :peek)
+
+    assert stack == ___
+  end
+
+  koan "Use cast to send an asynchronous request to the server" do
+    GenServer.start_link(ExampleGenServer, :ok, name: StackServer)
+
+    GenServer.cast(StackServer, {:push, 1})
+    GenServer.cast(StackServer, {:push, 3})
+    GenServer.cast(StackServer, {:push, 5})
```

Comment:
It would, yes. I agree that the demonstration of `sync`/`async` here is a bit sparse.
I will think of an example similar to `fast_process` and `slow_process` (but wrapped in a GenServer) from #180, with that, I should also get rid of that possibly confusing `:timer.sleep(10)`